src/mount/fuse/sfs_meta_fuse.cc

```diff
@@ -639,126 +657,124 @@ void sfs_meta_open(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info *fi) {
 		return;
 	}
 
-	status = fs_gettrashpath(ino,&path);
+	status = fs_gettrashpath(ino, &path);
 	status = saunafs_error_conv(status);
 
 	if (status) {
 		fuse_reply_err(req, status);
 	} else {
-		pathinfo = new pathbuf();
-
-		pathinfo->changed = 0;
-		pathinfo->size = strlen((char*)path) + 1;
-		if (!(pathinfo->p = (char*)malloc(pathinfo->size))) {
-			safs_pretty_syslog(LOG_EMERG, "out of memory");
-			delete pathinfo;
-			return;
-		}
-		memcpy(pathinfo->p, path, pathinfo->size - 1);
-		pathinfo->p[pathinfo->size - 1] = '\n';
+		pathinfo->changed = false;
+		pathinfo->size = strlen((char *)path) + 1;
```

Comment:
Avoid old C-Style cast operations.
```suggestion
		pathinfo->size = strlen(reinterpret_cast<const char *>(path)) + 1;
```

Reply:
Why not use std::string for path?

Reply:
the thing is that: `fs_gettrashpath` expects a `const uint8_t **` and returns a pointer to a C-style, null-terminated string buffer managed by the underlying code.
We cannot use std::string directly as an output parameter for this API.
Instead, we must first receive the result as a const uint8_t *, then convert it to std::string if needed. This ensures compatibility with the API and avoids unnecessary copies or type mismatches.